### PR TITLE
feat(types): added typeguards

### PR DESCRIPTION
**Pull request by michaelpautov**

### CODE REVIEW

File: src/schema/base.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        ROP: TokenSchema,
+        RIN: TokenSchema,
+        GOR: TokenSchema,
+        KOV: TokenSchema,
+        MUM: TokenSchema,
+        ARBT: TokenSchema,
+        OPTT: TokenSchema,
+        BSCT: TokenSchema,
+        HECT: TokenSchema,
+        ONET: TokenSchema,
```
Comment (maxklenk) [inline src/schema/base.ts:142]: All these lists in the schemas look very difficult to maintain. Can't we define one list with the key:id mapping and generate the lists for the schemas out of it?

File: src/types/step.ts [hunk 2]
```diff
@@ -101,33 +94,19 @@ export interface StepBase {
 
 export interface SwapStep extends StepBase {
   type: 'swap'
-  action: Action
```
Comment (maxklenk) [inline src/types/step.ts:104]: Why did you remove the `action` here?

File: src/types/step.ts [hunk 3]
```diff
@@ -101,33 +94,19 @@ export interface StepBase {
...
-  return step.type === 'cross'
-}
+export type Step = SwapStep | CrossStep | LifiStep
 
 export interface LifiStep extends StepBase {
   type: 'lifi'
   action: Action
   estimate: Estimate
   includedSteps: Step[]
 }
```
Comment (maxklenk) [inline src/types/step.ts:112]: The `includedSteps` in `LifiStep` won't be LifiSteps again. So we could remove the recursion here. Try creating
`export type IncludeStep = SwapStep | CrossStep` and add them to LifiStep instead.

**Review by maxklenk — CHANGES_REQUESTED** [review]
Great that you organized the types and constants better. I am not sure how much of the schemas is generated or manually configured by you.
I can't build or run the tests - so I am not sure what exactly you want me to review.

File: src/schema/base.ts (hunk 1 above)
Comment (michaelpautov) [reply src/schema/base.ts:142]: Will look at it.

File: src/types/step.ts (hunk 2 above)
Comment (michaelpautov) [reply src/types/step.ts:104]: Action is defined in StepBase interface

File: src/types/step.ts (hunk 3 above)
Comment (michaelpautov) [reply src/types/step.ts:112]: Thx
